src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -732,6 +732,18 @@ void VideoDecoder::maybeSeekToBeforeDesiredPts() {
   int firstActiveStreamIndex = *activeStreamIndices_.begin();
   const auto& firstStreamInfo = streams_[firstActiveStreamIndex];
   int64_t desiredPts = *maybeDesiredPts_ * firstStreamInfo.timeBase.den;
+
+  // For some encodings like H265, FFMPEG sometimes seeks past the point we
+  // set as the max_ts. So we use our own index to give it the exact pts we
+  // want.
```

Comment:
Correct me if I'm wrong - after the newly added `if` block, `desiredPts` isn't "the exact pts we want", right?